docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,325 @@
+```
+
+```json
+{
+  "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186111",
+  "Files": [
+    {
+      "Path": "/home/user/oras-install/sbom.spdx",
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "MediaType": "application/vnd.oci.image.manifest.v1+json",
+      "Digest": "sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "Size": 820,
+      "Annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+    },
+    {
+      "Path": "/home/user/oras-install/vul-scan.json",
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b",
+      "MediaType": "application/vnd.oci.image.manifest.v1+json",
+      "Digest": "sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b",
+      "Size": 820,
+      "Annotations": {
+        "org.opencontainers.image.title": "vul-scan.json"
+      }
+    }
+  ]
+}
+```
+
+> [!NOTE]
+> When pulling a folder to filesystem, the value of `Path` should be an absolute path of the folder and should be ended with slash `/` or backslash `\`, for example, `/home/Bob/sample-folder/` on Unix or `C:\Users\Bob\sample-folder\` on Windows. Other fields are the same as the example of pulling files as above.
+
+For example, pull an artifact that contains multiple layers (files) and show their descriptor metadata as raw JSON in the standard output.
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --format '{{toRawJson .}}'
+```
+
+```
+{"Ref":"localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186111","Files":[{"Path":"/home/user/oras-install/sbom.spdx","Ref":"localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222","MediaType":"application/vnd.oci.image.manifest.v1+json","Digest":"sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222","Size":820},{"Path":"/home/user/oras-install/vul-scan.json","Ref":"localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b","MediaType":"application/vnd.oci.image.manifest.v1+json","Digest":"sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b","Size":820}]}
+```
+
+### oras push
+
+Push two files to a repository and show the descriptor of the image manifest in pretty JSON format. 
+
+```bash
+oras push $REGISTRY/$REPO:$TAG sbom.spdx vul-scan.json --format json 
+```
+
+```json
+{
+  "Ref": "localhost:5000/oras@sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "MediaType": "application/vnd.oci.image.manifest.v1+json",
+  "Digest": "sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "Size": 820,
+  "Annotations": {
+    "org.opencontainers.image.created": "2023-12-15T09:41:54Z"
+  }
+}
+```
```

Comment:
Should we add `tag` values as well since we are pushing by tag?

Reply:
`tag` is an input value which is already known to user and doesn't change after oras commands executed. I think it is not necessary show a known value provided by user.

Reply:
Given the output has the tag in the regular output, I would suggest maybe a someting like 
`"refAsTag": "localhost:5000/oras:v1"`

- Enables clients to maybe log the json and understand that the tag was pushed with this digest and also only depend on json output in followin gcommands.

Reply:
> Enables clients to maybe log the json and understand that the tag was pushed with this digest and also only depend on json output in followin gcommands.

I would suggest add a `tags` field since multi-tagged pushing is supported

Reply:
Added `tags` to the properties as users may want to log the input value from the formatted output for auditing purposes.

Reply:
On further discussion would recommend a fully qualified array with a property maybe like `referenceByTags` 
I would however consider renaming it to `referenceByDigest`, even though obvious the fully qualified digest reference is singular an almost equivalent to an identifier, so would leave it as `ref` or `reference`. As that is the string that clients can use with confidence.